code/modules/mining/machine_input_output_plates.dm

```diff
@@ -1,19 +1,85 @@
+/obj/machinery/mineral
+	var/obj/machinery/hopper/input/input = null
+	var/obj/machinery/hopper/output/output = null
+	var/obj/machinery/mineralconsole/console = null

```

Comment:
These references aren't nulled in Destroy as far as I can see?

Reply:
They are, in `Unlink()`. I forgot console, though, good catch.